torchaudio/backend/_soundfile_backend.py

```diff
@@ -99,6 +100,8 @@ def load(
         channels_first (bool):
             When True, the returned Tensor has dimension ``[channel, time]``.
             Otherwise, the returned Tensor's dimension is ``[time, channel]``.
+        format (str, optional):
+            Not used. PySoundFile does not accept format hint.
```

Comment:
This makes me wonder whether it's time to introduce a top-level dispatch instead of using function pointers. Concretely right now we're doing

```torch.load = torchaudio.backend.sox_io_backend.load```

instead we could do

```
# torchaudio/__init__.py
def load(path):
    format = detect_format(path)
    [select backend]
    return backend.load(path, format)
```